tests/framework/results/helpers.py

```diff
@@ -39,30 +41,103 @@
 ##################
 
 
+class Hogwarts(Component):
+    @property
+    def columns_created(self) -> List[str]:
+        return [
+            "student_house",
+            "familiar",
+            "power_level",
+            "previous_house_points",
+            "house_points",
+            "previous_quidditch_wins",
+            "quidditch_wins",
+        ]
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        size = len(pop_data.index)
+        rng = np.random.default_rng(42)
+        initialization_data = pd.DataFrame(
+            {
+                "student_house": rng.choice(STUDENT_HOUSES, size=size),
+                "familiar": rng.choice(FAMILIARS, size=size),
+                "power_level": rng.choice([str(lvl) for lvl in POWER_LEVELS], size=size),
+                "previous_house_points": 0,
+                "house_points": 0,
+                "previous_quidditch_wins": 0,
+                "quidditch_wins": 0,
+            },
+            index=pop_data.index,
+        )
+        self.population_view.update(initialization_data)
+
+    def on_time_step(self, pop_data: SimulantData) -> None:
+        update = self.population_view.get(pop_data.index)
+        update["previous_house_points"] = update["house_points"]
+        update["previous_quidditch_wins"] = update["quidditch_wins"]
+        # House points are stratified by 'student_house' and 'power_level'.
+        # Let's have each wizard of gryffindor and of level 50 and 80 gain a point
+        update.loc[
+            (update["student_house"] == "gryffindor")
+            & (update["power_level"].isin(["50", "80"])),
+            "house_points",
+        ] += 1
```

Comment:
Per discussions w/ Rajan: change this to NOT increase (i.e. be the points PER TIME STEP) and then we can just use `sum()` as the aggregator. This is more in line with how we do things